plugins/slack_webhooks/app/assets/javascripts/slack_webhooks/application.js

```diff
@@ -0,0 +1,111 @@
+//= require jquery_mentions_input/jquery.elastic.source.js
+//= require jquery_mentions_input/jquery.mentionsInput.js
+
+samson.factory('SlackMentionbox', function ($rootScope, Slack) {
+  var self = this;
+  self.users = Slack.users();
+
+  self.filteredData = function (query) {
+    return _.filter(self.users, function (item) {
+      return item.name.toLowerCase().indexOf(query.toLowerCase()) > -1;
+    });
+  };
+
+  self.markupData = function (callback) {
+    $(self.mentionsId).mentionsInput('val', function(text) {
+      text = self.reformatMessage(text);
+      callback(text);
+    });
+  };
+
+  self.reformatMessage = function(text) {
+    return text.replace(/@\[([^\\]+?)\]\(([^)]+?)\)/g, "@$1");
+  };
+
+  $rootScope.$on('slack_webhooks_users', function () {
+    self.draw();
+  });
+
+  self.draw = function() {
+    $(self.mentionsId).mentionsInput({
+      defaultValue: self.defaultMessage,
+      onDataRequest:function (mode, query, callback) {
+        callback.call(this, self.filteredData(query));
+      }
+    });
+  };
+
+  self.init = function(id, defaultMessage) {
+    self.mentionsId = id;
+    self.defaultMessage = defaultMessage;
+    self.draw();
+    return {
+      message: self.markupData
+    };
+  };
+
+  return self;
+});
+
+samson.factory('Slack', function ($rootScope, $http) {
+  var self = this;
+
+  self.users = function () {
+    var users = [];
+
+    $http.get('/slack_webhooks/users').success(function (data) {
+      users.push.apply(users, data.users);
+      $rootScope.$emit('slack_webhooks_users', users);
+    }).error(function () {
+      console.log('Could not fetch the Slack users!');
+    });
+
+    return users;
+  };
+
+  self.buddyRequest = function (deploy, message) {
+    return $http.post('/slack_webhooks/notify', { deploy_id: deploy, message: message });
```

Comment:
prefer reading urls from data-xyz attributes so they do not get hardcoded in js land